http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java

```diff
@@ -406,6 +406,7 @@ public DefaultHttpClient(@Nullable LoadBalancer loadBalancer,
 
         Optional<Duration> connectTtl = configuration.getConnectTtl();
         this.connectionTimeAliveMillis = connectTtl.map(duration -> !duration.isNegative() ? duration.toMillis() : null).orElse(null);
+        final ChannelHealthChecker channelHealthChecker = channel -> channel.eventLoop().newSucceededFuture(channel.isActive() && !ConnectTTLHandler.isChannelExpired(channel));
```

Comment:
@yawkat Here is where the expiration check is added. Expired channels won't be considered healthy, and then they are closed in SimpleChannelPool 'notifyHealthCheck' as a result of failing the health check.

Reply:
i understand the idea, but `notifyHealthCheck` is only called from `doHealthCheck`, which is only called during acquire. However if the health check fails in `doHealthCheckOnRelease` because the channel has expired, then it won't be returned to the pool at all, and `notifyHealthCheck` will never be called.

Reply:
ideally the channel closure would be asserted by tests, but it might not be feasible to write a test for this atm, maybe once there is EmbeddedChannel support for the client.

Reply:
oh. I see. releaseAndOfferIfHealthy does not close the channel if it is unhealthy.

Reply:
There are tests in ConnectionTTLSpec. I updated that earlier to cover the scenario where the connection expires while in the pool. I have checked and can get a test to fail for the returning to pool case. I will push some changes shortly.

Reply:
ah awesome, didn't see those tests. I'll look at your updates tomorrow.